cla_backend/apps/cla_auth/auth.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import sys
+import traceback
+
+from django.conf import settings
+
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+
+from provider.oauth2.models import Client
+
+from mohawk import Receiver
+from mohawk.exc import HawkFail, TokenExpired
+from mohawk.util import parse_authorization_header
+
+from hawkrest import seen_nonce
+
+log = logging.getLogger(__name__)
+# Number of seconds until a Hawkmessage expires.
+default_message_expiration = 60
+
+
+class OBIEEHawkAuthentication(BaseAuthentication):
+    def authenticate(self, request):
+        if not request.META.get('HTTP_AUTHORIZATION'):
+            log.debug('request did not send an Authorization header')
+            raise AuthenticationFailed('missing authorization header')
```

Comment:
Not sure you should raise `AuthenticationFailed`. Based on the docs, you should always `a two-tuple of (user, auth) if authentication succeeds, or None otherwise.`
Might be wrong though.